Content.Server/Ninja/Systems/NinjaSuitSystem.cs

```diff
@@ -0,0 +1,155 @@
+            return;
+
+        // TODO: do actions like with gloves are now (before merge)
+        _actions.AddAction(user, comp.TogglePhaseCloakAction, uid, actions);
+        _actions.AddAction(user, comp.RecallKatanaAction, uid, actions);
+        // TODO: ninja stars instead of soap, when embedding is a thing
+        _actions.AddAction(user, comp.CreateSoapAction, uid, actions);
+        _actions.AddAction(user, comp.KatanaDashAction, uid, actions);
+        _actions.AddAction(user, comp.EmpAction, uid, actions);
+    }
+
+    // TODO: put in shared so client properly predicts insertion, but it uses powercell so how???
+    private void OnSuitInsertAttempt(EntityUid uid, NinjaSuitComponent comp, ContainerIsInsertingAttemptEvent args)
+    {
+        // no power cell for some reason??? allow it
+        if (!_powerCell.TryGetBatteryFromSlot(uid, out var battery))
+            return;
+
+        // can only upgrade power cell, not swap to recharge instantly otherwise ninja could just swap batteries with flashlights in maints for easy power
+        if (!TryComp<BatteryComponent>(args.EntityUid, out var inserting) || inserting.MaxCharge <= battery.MaxCharge)
+        {
+            args.Cancel();
+        }
+    }
+
+    protected override void UserUnequippedSuit(EntityUid uid, NinjaSuitComponent comp, EntityUid user)
+    {
+        base.UserUnequippedSuit(uid, comp, user);
+
+        // remove suit ability actions
+        _actions.RemoveProvidedActions(user, uid);
+
+        // remove power indicator
+        _ninja.SetSuitPowerAlert(user);
+    }
+
+    private void OnTogglePhaseCloak(EntityUid uid, NinjaSuitComponent comp, TogglePhaseCloakEvent args)
+    {
+        args.Handled = true;
+        var user = args.Performer;
+        // need 1 second of charge to turn on stealth
+        var chargeNeeded = SuitWattage(comp);
+        if (!comp.Cloaked && (!_ninja.GetNinjaBattery(user, out var battery) || battery.CurrentCharge < chargeNeeded || _useDelay.ActiveDelay(uid)))
+        {
+            _popups.PopupEntity(Loc.GetString("ninja-no-power"), user, user);
+            return;
+        }
+
+        comp.Cloaked = !comp.Cloaked;
+        SetCloaked(args.Performer, comp.Cloaked);
+        RaiseNetworkEvent(new SetCloakedMessage()
+        {
+            User = user,
+            Cloaked = comp.Cloaked
+        });
+    }
+
+    private void OnCreateSoap(EntityUid uid, NinjaSuitComponent comp, CreateSoapEvent args)
+    {
+        args.Handled = true;
+        var user = args.Performer;
+        if (!_ninja.TryUseCharge(user, comp.SoapCharge) || _useDelay.ActiveDelay(uid))
+        {
+            _popups.PopupEntity(Loc.GetString("ninja-no-power"), user, user);
+            return;
+        }
+
+        // try to put soap in hand, otherwise it goes on the ground
+        var soap = Spawn(comp.SoapPrototype, Transform(user).Coordinates);
+        _hands.TryPickupAnyHand(user, soap);
+    }
```

Comment:
While this is hilarious, it's a bit out of place. The cooldown should be exceptionally longer otherwise the ninja can just flood the entire hall with soap, even though the soap does disappear after awhile and does cost power.

Reply:
I'll up it to 10 seconds then